### ISSUE TITLE
FAQ: What is new with session rollovers in RC 0.10.4?

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/pocket-core#1577]
## Objective

Follow up on the session rollover feature from [RC 0.10](https://www.notion.so/pocketnetwork/RC-0-10-0-Release-Guide-938ecbf8f69f40e1bdb92342084de52c?pvs=4).

## Origin Document

- [PIP-28: DAN - Distributing the Altruist Network (Community Chains Introduction)][pip-28]
- [RFP/Improvement - Allowing for configurable session block height tolerance](https://github.com/pokt-network/pocket-core/issues/1464)
- [Release Notes for Session Rollover in RC 0.10](https://www.notion.so/pocketnetwork/RC-0-10-0-Release-Guide-938ecbf8f69f40e1bdb92342084de52c?pvs=4#5447d1b3ce8a4365b33b463d99ce0045)
- #1536

## Goals

- Gain insight into session rollover adoption
- Understand to what degree Session Rollover eliminate the need of altruists
- Educate the node runners on how/why to enable session rollovers

## Deliverable

- [ ] 1. Work with the portal backend/infra team to collect analytics on how altruists are being used (cc @kutoft @fredteumer)
- [ ] 2. Send out a survey to all node runners to understand what value everyone has set `client_session_sync_allowance` to in their configurations
- [ ] 3. Put together a 1 pager on what node runners should set `client_session_sync_allowance` to in order to enable session rollover
	- [ ] **Optional**: 3.1 Work with the `poktpool` team to translate this into an infographic
- 4. After all of the above are done:
	- [ ] 4.1 Spearhead adoption of `client_session_sync_allowance` from `0` to `1`
	- [ ] 4.2 Track the impact of `(4.1)` on `(1)` as adoption increases
	- [ ] 4.3 Evaluate the network cost savings (see [pip-28][pip-28]) based on the result of `(4.2)`

## Non-goals / Non-deliverables

- Implementing new features or business logic
- Forcing all node runners to adopt session rollovers
- Eliminating the need for altruists

---

**Creator**: @Olshansk 
**Co-Owners**: @RossiNYC 

[pip-28]: https://forum.pokt.network/t/pip-28-dan-distributing-the-altruist-network-community-chains-introduction/4020

**Comment by RossiNYC** [comment]
@Olshansk - is there a way to proactively view this and other parameters/settings w/o having to survey node runners directly? 

> Send out a survey to all node runners to understand what value everyone has set _client_session_sync_allowance_ to in their configurations

**Comment by Olshansk** [comment]
> is there a way to proactively view this and other parameters/settings w/o having to survey node runners directly?

**tl;dr No - not to my knowledge.**

---

_Extra Unecessary Details & Investigation_

###  Governance Params

I was able to query governance params like so

```bash
export POCKET_ENDPOINT=https://mainnet.gateway.pokt.network/v1/lb/60a2ac11b1747c6552385c61
pocket query params --remoteCLIURL ${POCKET_ENDPOINT}
```

Which  gives:

```json
{
    "app_params": [
        {
            "param_key": "application/MaximumChains",
            "param_value": "15"
        },
        ...
```

But this does not give node configurations....

### Tendermint Status / Configs

Since these are [Tendermint level configurations](https://docs.tendermint.com/v0.33/tendermint-core/configuration.html), I tried to, I tried to query the status of the [Tendermint notes on TesetNet](https://github.com/pokt-network/pocket-core/blob/staging/doc/guides/testnet.md) provided by the NodeFleet team (cc @StevenMartinez94 @aqt01) like so:

```bash
curl -S "https://testnet:${NODE_FLEET_PASSWORD}@node1.tendermint.testnet.pokt.network/status"
```

But this does not show node-specific configurations:

```json
{
  "jsonrpc": "2.0",
  "id": -1,
  "result": {
    "node_info": {
      "protocol_version": {
        "p2p": "7",
        "block": "10",
        "app": "0"
      },
      "id": "77e608d8ae4cd7b812f122dc82537e79dd3565cb",
      "listen_addr": "tcp://86.48.21.90:26662",
      "network": "testnet",
      "version": "0.33.8",
      "channels": "4020212223303800",
      "moniker": "testnet-12",
      "other": {
        "tx_index": "on",
        "rpc_address": "tcp://0.0.0.0:26657"
      }
    },
    "sync_info": {
      "latest_block_hash": "9501F2C0A01C80C99B678A9D4B085439D6AAA55D69BAC8FAD0F4E6E02C6502E4",
      "latest_app_hash": "29DAB7A057F099C0F813ACBFC0CAB7654298409F3C3C0241F7164F21F633E812",
      "latest_block_height": "113790"
```

### Node Configs

I ended up asking ChatGPT if it is aware of a way to query node-specific configurations (i.e. the off-chain, non-governance, node specific configs) and it also told me there's no way:

![Screenshot 2023-08-03 at 11 34 11 AM](https://github.com/pokt-network/pocket-core/assets/1892194/f20f720c-60a1-4293-817d-431e602604f7)

---

In short, unless the whole network (and forks) adopts a software change that exposes a new endpoint to expose this, I'm personally not aware of a way to do this.

**Comment by Olshansk** [comment]
@RossiNYC I wanted to know:

1. Was the context above sufficient?
2. Any progress on this issue?
3. Any blockers or support you need on this?

**Comment by RossiNYC** [comment]
@Olshansk - set up call with @fredteumer & @kutoft for tomorrow afternoon, there was some concern in the last PNF All Hands around over-surveying so I was going to bring this up tomorrow morning on the next call which is why I was asking if the data was available proactively - I'm sure we can get the info, will get feedback/greenlight from them tomorrow (unless I confused a few things here and this isn't the proper process?) Let me know if I am over thinking it!

**Comment by Olshansk** [comment]
> @Olshansk - set up call with @fredteumer & @kutoft for tomorrow afternoon, there was some concern in the last PNF All Hands around over-surveying so I was going to bring this up tomorrow morning on the next call which is why I was asking if the data was available proactively - I'm sure we can get the info, will get feedback/greenlight from them tomorrow (unless I confused a few things here and this isn't the proper process?) Let me know if I am over thinking it!

Sounds like a great plan!
